### PR TITLE
Racon wrapper

### PR DESCRIPTION
**Pull request by mmiladi**

### CODE REVIEW

File: tools/racon/racon.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        <param type="data" name="reads" format="fasta,fasta.gz,fastq,fastq.gz" label="sequences"/>
+        <param type="data" name="overlaps" format="sam,sam.gz,paf,paf.gz,mhap,mhap.gz" label="overlaps"/>
+        <param type="data" name="corrected_reads" format="fasta,fasta.gz,fastq,fastq.gz" label="target sequences"/>
+    </inputs>
+    <outputs>
+        <data name="consensus" format="fasta" from_work_dir="racon_polished_consensus.fa" label="Consensus " />
+    </outputs>
+    <tests>
+        <test>
+            <param name="reads" ftype="fasta" value="test-data/sample_reads.fasta"/>
```
Comment (bgruening) [inline tools/racon/racon.xml]: you do not need test-data/ here ....

File: tools/racon/racon.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    <command detect_errors="exit_code"><![CDATA[
+        racon 
+            '$reads' 
+            '$overlaps' 
+            '$corrected_reads'
+            > racon_polished_consensus.fa  
+    ]]></command>
+    <inputs>
+        <param type="data" name="reads" format="fasta,fasta.gz,fastq,fastq.gz" label="sequences"/>
+        <param type="data" name="overlaps" format="sam,sam.gz,paf,paf.gz,mhap,mhap.gz" label="overlaps"/>
```
Comment (bgruening) [inline tools/racon/racon.xml]: does Galaxy know paf and mhap?

File: tools/racon/racon.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+<tool id="racon" name="Consensus module for raw de novo DNA assembly of long uncorrected reads." version="0.1.0">
+    <macros>
```
Comment (bgruening) [inline tools/racon/racon.xml]: description is missing
version_command is missing

File: tools/racon/racon.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
+<tool id="racon" name="Consensus module for raw de novo DNA assembly of long uncorrected reads." version="0.1.0">
```
Comment (bgruening) [inline tools/racon/racon.xml]: version 1.3.1?

File: tools/racon/macros.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <requirements>
+        <requirement type="package" version="1.3.1">racon</requirement>
+            <yield/>
+        </requirements>
+    </xml>
+    <xml name="citations">
+        <citations>
+            <citation type="doi">10.1101/gr.214270.116</citation>
+            <citation type="bibtex">
+@misc{githubracon,
```
Comment (bgruening) [inline tools/racon/macros.xml]: the bibtex can be remove if you have a DOI

File: tools/racon/racon.xml (hunk 2 above)
Comment (mmiladi) [reply tools/racon/racon.xml]: Don't think so.

File: tools/racon/macros.xml (hunk 5 above)
Comment (mmiladi) [reply tools/racon/macros.xml]: bibtex is for the github link

File: tools/racon/racon.xml (hunk 1 above)
Comment (mmiladi) [reply tools/racon/racon.xml]: :+1: thanks

File: tools/racon/racon.xml (hunk 2 above)
Comment (bgruening) [reply tools/racon/racon.xml]: Then I would remove it.

File: tools/racon/macros.xml (hunk 5 above)
Comment (bgruening) [reply tools/racon/macros.xml]: Yes, but we assume this is mentioned in the paper.

**Comment by mmiladi** [comment]
@bgruening can u please try to update test result on your system? On my Ubuntu laptop anaconda racon binary crashes with exception `Illegal instruction (core dumped)`. On travis it seems to work fine!

**Comment by bgruening** [comment]
I can, if you get a bike in the next 15min ;)

**Comment by mmiladi** [comment]
:-D :+1:  Ready to merge :P

File: tools/racon/test-data/consensus_result1.fasta [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
...ACTTTCACGGTAGTTTTTGCCGCTTTACCGCCCAGGCAGCCCAGAAGCCACGCCCATTAGTGAAACGCTTCATGGTGAGCGTGTTATCCCGGTGCTTTTGCCATACCACGGGGCCAGCGCCAGCAGCGACGGAATATCACGAATAGTCGGCTCAACGTGGGTTTTCATAAAGTTCTCGGCATCACCATCCGTCGGCAACCAGATAAGGGTGTTGCGCTGCTTATGCTCTATAAAGTAGGCATAAACACCCAGCAGCATTTTGGAATAACCGACACGGGCAGACTTCACCACATTCACCTCACGGATGTAGTCGCTGCCATCGCATTCATGATGGCCCGCTGAAAGGCAGTGTTTCCCAGCGCCCTTCCTGGTATGCGGATTCTTTCGGGGAGATAGTAATTAGCATCCGCCCATTCAACGGCGGTCTGTGGCTCCGGCCTGAACAGTGAGCGAAGCCCGGCGCGGACAAAATGCCGCAGCCTGTTAACCTGACGACTGTCGATATATTCACTGAGCAACCCCGGTATCAGTTCATCCAGCGCGGCTGCTTTGTTCATGGCTTTGATGATATCCCGTTTCAGGAAATCAACATGTCGGTTTTCCAGTTCCGGAAAACGCCGCTGCACCGACAGGGGGAGCCGTCGAGAATACTGGCAATTTCACCTGCGATCCGCGACAGCACGAAAGTACAGAATGCGGTTTCCACCACTTCAGCGGAGTCTCTGGCATTCTTCAGTTCCTGTGCGTCGGCCTGGGCACGCGTAAGTCGATGGCGTTCGTACTCAATAGTTCCTGGCTGGAGATCTGCCTCGCTGCCTGCCTGCCGCAGTTCTTCAACCTCCCGGCGCAGCTTTTCGTTCTCAATTTCAGCATCCCTTCGGCATACCATTTTATGACGCTGCAGAGTCATAAAGCACCTCATTACCCTTGCCACCGCCTCGCAGAACGGGCATTCCCTGTTCCTGCCAGTTCTGAATGGTACGGATACTCGCACCGAAAATGTCAGCCAGCTGCTTTTGTTGACTTCCATTGTTCATTCCACGGACAAAACAGAGAAAGGAACGACAGAGGCCAAAAGCTCGCTTTCAGCAGCCCTGTCGTTTCCTTCTTTCAGAGGGTATTTTAAATAAAACATTAAGTTATGACGAAGAAGAACGGAAACGCCTTAAACCGGAAAATTTTCATAAATAGCGAAAACCC
```
Comment (bgruening) [inline tools/racon/test-data/consensus_result1.fasta]: this test file is not used ...

**Comment by bgruening** [comment]
I thought racon is crashing on your instance and needs to be recompiled?

**Comment by mmiladi** [comment]
@@bgruening Opps sorry, test1 can be uncommented. If u would like extra tests :)

**Comment by mmiladi** [comment]
@bgruening crashing is due to the conda binary. Nothing can be done here :) right?

**Comment by bgruening** [comment]
> @bgruening crashing is due to the conda binary. Nothing can be done here :) right?

Yes, but if people install this tool and it crashes we will get error reports and complains. Also we can not install it :)

**Comment by mmiladi** [comment]
@bgruening Спасибо ! :)

**Pull request merged**